packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -0,0 +1,428 @@
+
+contract NounsAuctionHouseV2 is
+    INounsAuctionHouseV2,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable
+{
+    /// @notice A hard-coded cap on time buffer to prevent accidental auction disabling if set with a very high value.
+    uint56 public constant MAX_TIME_BUFFER = 1 days;
+
+    /// @notice The Nouns ERC721 token contract
+    INounsToken public immutable nouns;
+
+    /// @notice The address of the WETH contract
+    address public immutable weth;
+
+    /// @notice The duration of a single auction
+    uint256 public immutable duration;
+
+    /// @notice The minimum price accepted in an auction
+    uint192 public reservePrice;
+
+    /// @notice The minimum amount of time left in an auction after a new bid is created
+    uint56 public timeBuffer;
+
+    /// @notice The minimum percentage difference between the last bid amount and the current bid
+    uint8 public minBidIncrementPercentage;
+
+    /// @notice The active auction
+    INounsAuctionHouseV2.AuctionV2 public auction;
+
+    /// @notice Whether this contract is paused or not
+    /// @dev Replaces the state variable from PausableUpgradeable, to bit pack this bool with `auction` and save gas
+    bool public __paused;
+
+    /// @notice The Nouns price feed state
+    mapping(uint256 => SettlementState) settlementHistory;
+
+    constructor(
+        INounsToken _nouns,
+        address _weth,
+        uint256 _duration
+    ) {
+        nouns = _nouns;
+        weth = _weth;
+        duration = _duration;
+    }
+
+    /**
+     * @notice Initialize the auction house and base contracts,
+     * populate configuration values, and pause the contract.
+     * @dev This function can only be called once.
+     */
+    function initialize(
+        uint192 _reservePrice,
+        uint56 _timeBuffer,
+        uint8 _minBidIncrementPercentage
+    ) external initializer {
+        __Pausable_init();
+        __ReentrancyGuard_init();
+        __Ownable_init();
+
+        _pause();
+
+        reservePrice = _reservePrice;
+        timeBuffer = _timeBuffer;
+        minBidIncrementPercentage = _minBidIncrementPercentage;
+    }
+
+    /**
+     * @notice Settle the current auction, mint a new Noun, and put it up for auction.
+     */
+    function settleCurrentAndCreateNewAuction() external override whenNotPaused {
+        _settleAuction();
+        _createAuction();
+    }
+
+    /**
+     * @notice Settle the current auction.
+     * @dev This function can only be called when the contract is paused.
+     */
+    function settleAuction() external override whenPaused {
+        _settleAuction();
+    }
+
+    /**
+     * @notice Create a bid for a Noun, with a given amount.
+     * @dev This contract only accepts payment in ETH.
+     */
+    function createBid(uint256 nounId) external payable override {
+        INounsAuctionHouseV2.AuctionV2 memory _auction = auction;
+
+        if (_auction.nounId != nounId) revert NounNotUpForAuction();
```

Comment:
@eladmallel I've recently started preferring string errors and `require` instead of custom errors. often it's easier to debug with string errors.
what do we gain from the custom errors? I don't think contract size is an issue here.
it would also reduce the diff and external interface compared to AuctionHouseV1

wdyt?

Reply:
fixed: https://github.com/nounsDAO/nouns-monorepo/pull/799/commits/79f611a6de8323c8ba7e1063087a4c99eb3e4a80